common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -363,6 +365,95 @@ static int parseRoutes(struct nlmsghdr *nlHdr, std::shared_ptr<route_info> rtInf
     return (0);
 }
 
+std::shared_ptr<std::unordered_map<std::string, std::string>>
+network_utils::get_arp_table(bool mac_as_key)
+{
+    auto arp_table_ptr = std::make_shared<std::unordered_map<std::string, std::string>>();
+    auto arp_table     = *arp_table_ptr.get();
+
+    // Allocate a new netlink socket
+    auto nl_socket = nl_socket_alloc();
+    if (nl_socket == nullptr) {
```

Comment:
```suggestion
    if (!nl_socket) {
```

Reply:
We usually require it only for smart pointer because of a Klocwork error.
But ok, I'll change it.